lib/rules/switch-final-break.js

```diff
@@ -0,0 +1,189 @@
+            switchFinalBreakAlways:
+                "Final clause in 'switch' statement should end with 'break;'."
+        },
+        schema: [
+            {
+                enum: ["never", "always"]
+            }
+        ],
+        type: "suggestion"
+    },
+    create(context) {
+        let currentCodePath = null;
+        const requireBreak = context.options[0] ? context.options[0] : "never";
+
+        /**
+         * Returns the statements list of a given switch clause
+         * @param {ASTNode} node switch clause node
+         * @returns {Array} - the statements list of the switch clause
+         */
+        function getStatements(node) {
+            if (
+                node.consequent.length === 1 &&
+                node.consequent[0].type === "BlockStatement"
+            ) {
+
+                // block statement
+                const block = node.consequent[0];
+
+                return block.body;
+            }
+            return node.consequent;
+        }
+
+        /**
+         * Returns the last statement of a given switch clause
+         * @param {ASTNode} node switch clause node
+         * @returns {ASTNode} - the final statement of the switch clause
+         */
+        function getLastStatement(node) {
+            return lodash.last(getStatements(node));
+        }
+
+        /**
+         * Checks whether or not a given code path segment is reachable.
+         * @param {CodePathSegment} segment A CodePathSegment to check.
+         * @returns {boolean} `true` if the segment is reachable.
+         */
+        function isReachable(segment) {
+            return segment.reachable;
+        }
+
+
+        return {
+            onCodePathStart(codePath) {
+                currentCodePath = codePath;
+            },
+            onCodePathEnd() {
+                currentCodePath = currentCodePath.upper;
+            },
+
+            "SwitchCase:exit"(node) {
+                if (requireBreak !== "always") {
+                    return;
+                }
+                if (node.type === "SwitchCase" && node.consequent.length === 0) {
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => fixer.insertTextAfter(
+                            node,
+                            " break;"
+                        )
+                    });
+                }
+                if (currentCodePath.currentSegments.some(isReachable) &&
+                    node.consequent.length > 0 &&
+                    lodash.last(node.parent.cases) === node
+                ) {
+                    const lastStatement = getLastStatement(node);
+
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => {
+                            const lastStatementText = context.getSourceCode().getLines()[
```

Comment:
is there some guarantee that `indent` fixer will run after this rule fixer (so users won't need to run fixers twice)?
also in case the user don't have a rule that fixes indentations, are you ok with not preserving their indent?

Reply:
Yes - the autofixer does multiple passes when it runs. As a general rule of thumb, we try to keep autofixers as small as possible, and this change would be in step with that.